### DISCUSSION TITLE
Welcome to vf1 Discussions!

### DISCUSSION THREAD

**Discussion by imarinfr** [imarinfr/vf1#5] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by huchzi** [comment]
Thank you for your efforts with the visualFields package,  Ivan.

What would you personally consider the main purpose of the package? What are people using it for?

I'm using it to import visual field data and to perform statistical analysis on these data. I'm interested in functions that provide methods for statistical analyses, for example PoPLR. I don't really need the plotting capabilities, especially because they work poorly with Octopus fields.

Do you know anybody who uses the package for..
- visualizing fields from commercial perimeters?
- visualizing results from experimental perimeters?
- analyzing/visualizing data from tests created with the Open Perimetry Interface?
- other purposes?

How many people are working with the package that you are aware of?

What do you consider the most important tasks for further development of the package? Maybe I could contribute something...

**Reply by imarinfr** [reply]
Thank you for your comment.

One of the most important aims of the Open Perimetry Initiative, of which visualFields is part of, is to aid researchers transition to  open science, where there are no black boxes on their methods that are otherwise inescapable when using proprietary software.

I think most people use it for the implementations of some visual field indices that are not easily accessible otherwise. The PoPLR implementation is a prime example. More experimental labs use it to generate their own normative data for analysis. This is particularly useful to test new perimetry devices against stablished ones. Using the same group of healthy people for the development of device-specific normative values reduces the risk of unwanted, sample dependent, biases.

There are about 10 labs that have used or are currently using the package that I am aware of. The use is as you describe, analysing fields from commercial perimeters and compare them against experimental ones (with the same methods and data, so that the only variable is hardware). The experimental perimeters use the OPI.

Regarding the future of the package, the following is in my wish list, all related to making visualFields more accessible:

1. Generate a good guide and sample code. There is some done, but it is sparse. A vignette in a JSS style (https://www.jstatsoft.org/index) would be very useful. I envision this vignette starting with perimetry theory and then explaining how it is implemented in visualFields.
2. Add demos and graphical tools for new users have a more solid/appealing starting point.
3. Extend the visualFields usability to micro-perimetry and structure-driven perimetry. This probably involves the generation of another package for imaging processing and an add-on for SF analyses.

If you have any suggestions for contribution, I would be very happy to hear about them.
